File: Andy/src/level2/java/n2arraysplit.java
package level2.java;

// 프로그래머스 Level 2 ,n^2 배열 자르기
public class n2arraysplit {
    public int[] solution(int n, long left, long right) {
        int[] answer = new int[(int)right-(int)left+1];

        long idx = 0;
        int i = 0;
        for(long x=1;x<=n;x++) {
            if(idx>right) break;
            if(idx+n < left) {
                idx += n;
                continue;
            }
            for(long y=1;y<=n;y++) {
                long d = y<=x?x:y;            
                if(idx>=left && idx<=right) answer[i++] = (int)d;
                idx++;
            }
        }

        return answer;
    }
}

File: Andy/src/level2/javascript/n2arraysplit.js
// 프로그래머스 Level 2 ,n^2 배열 자르기
function solution(n, left, right) {
    var answer = [];
    var arr = [];
    var idx = 0;
    let x = parseInt(left/n)+1;
    idx = n * (x-1);
    
    for(;x<=n;x++) {
        if(idx>right) break;
        if(idx+n < left) {
            idx += n;
            continue;
        }

        for(let y=1;y<=n;y++) {
            let d = y<=x?x:y;            
            if(idx>=left && idx<=right) answer.push(d);
            idx++;
        }
    }
    return answer;
}